infra/modules/azure_monorepo_single_env_starter_pack/github_branch_rules.tf

```diff
@@ -0,0 +1,25 @@
+resource "github_branch_default" "main" {
+  repository = github_repository.this.name
+  branch     = "main"
+}
+
+resource "github_branch_protection" "main" {
+  repository_id = github_repository.this.name
+  pattern       = "main"
+
+  required_status_checks {
+    strict   = false
+    contexts = []
+  }
+
+  require_conversation_resolution = true
+  require_signed_commits          = false
+
+  required_pull_request_reviews {
+    dismiss_stale_reviews           = false
+    require_code_owner_reviews      = true
+    required_approving_review_count = 1
+  }
+
```

Comment:
```suggestion
# Restrict who can push to protected branches
  push_restrictions = [
    "admins",
    data.github_organization_teams.all.teams["maintainers"].node_id
  ]
```

Reply:
You sure? When admins should be able to push on the main branch? I see that occurring by mistake very often